Minor improvements for retail best practices (#2392)
uyuni-project/uyuni-docs: modules/retail/pages/retail-best-practices.adoc

[[retail-best-practices]]
= Best practices

== Deploying new images

This file describes how to deploy a new image on selected terminals first, test it, and then deploy it to all terminals.

> The hardware type group [literal]``HWTYPE`` is used for booting new terminals, while already deployed terminals can be moved to any group that provides the {saltboot} formula. 

Reviewer: This is sending different message then originally thought.
The intent was to provide some recognizable string for hardware type group.

In the new version it seems like we are talking about hardware type group named `HWTYPE` which is not the case. `HWTYPE` is just a prefix for the hardware type group name.

> The hardware type group [literal]``HWTYPE`` is used for booting new terminals, while already deployed terminals can be moved to any group that provides the {saltboot} formula. 

Author: ```suggestion
The hardware type group (e.g [literal]``HWTYPE``) is used for booting new terminals, while already deployed terminals can be moved to any group that provides the {saltboot} formula. 
```

This flexibility allows for testing of new images or configurations. 

//To ensure a smooth deployment, follow these steps:

> .Procedure: Smooth image deployment

Reviewer: Smooth is interesting choice 😃 

This procedure is how one can have two groups controlling the image deployment. In practice this is mimicking A/B image deployment. So we can call this either `A/B image deployment`, or `Dual group image deployment`.

> .Procedure: Smooth image deployment

Author: ```suggestion
.Procedure: Dual group image deployment
```

. Build new image.
    Use image synchronization formula to synchronize the image to all Branch Servers.
. Create a test group.
  * Name the group, for example, [literal]``TEST:POSVendor-Terminal``. The name can be chosen freely.

>   * Add the {saltboot} formula to this group from the original hardware type group. 

Reviewer 2: I am not sure if this is clear enough.
They have to manually enter the same values again.

Author: @nadvornik -- can you please add text that would correctly reflect that?

Reviewer 2: Something like: Add the {saltboot} formula to this group. Use the same configuration as in the original HWTYPE group, except for the image version. Use the name and version of the new image.

>   * Add the {saltboot} formula to this group, replicating the configuration from the original [literal]``HWTYPE`` group, except for the image version. 

Reviewer: Maybe I am reading it wrong, but I would just write `from the original hardware type group`.

>   * Add the {saltboot} formula to this group, replicating the configuration from the original [literal]``HWTYPE`` group, except for the image version. 

Author: ```suggestion
  * Add the {saltboot} formula to this group from the original hardware type group. 
```

  * Transfer these terminals from the group [literal]``HWTYPE:POSVendor-Terminal`` group to the newly created group [literal]``TEST:POSVendor-Terminal``. 
    It is recommended to use System Set Manager for this task.
. Reboot terminals.
  * Power on the selected POS terminals. 
    They should boot the new image.
. Verify and evaluate.
  * Monitor the terminals in the [literal]``TEST:POSVendor-Terminal`` group.
  * Check if the new image performs as expected and if there are any issues or errors encountered.
. Update [literal]``HWTYPE`` group.
  * Once you have confirmed that the new image works correctly, the [literal]``HWTYPE`` group can be updated.
  * Modify the image version in the [literal]``HWTYPE:POSVendor-Terminal`` group to match the tested and approved version.
  * This ensures that all other terminals receive the new image.

=== Controlling image versions

By default, {saltboot} formula selects the specified image name and version or selects the highest one if the version is not specified.

There are multiple methods to control the image version within the [literal]``HWTYPE`` and [literal]``TEST``` groups:

> .Procedure: Image control

Reviewer: This paragraph is not a procedure, but rather list of options how can one control image deployment.
* specifying exact versions in A/B group created above
* using inactive flag to deactivate image
* using freeze flag to disable image autoupdate

* Specify exact version.
** In both the [literal]``HWTYPE`` and [literal]``TEST`` groups, specify explicitly the exact version of the image to be deployed.
** Make sure that the specified image version is already synchronized.
* Mark new image as [literal]``Inactive``.
** In the [literal]``HWTYPE`` group, do not specify the image version.
** Mark the new image as [literal]``Inactive`` after building and before synchronizing it to Branch Server.
** The [literal]``inactive`` flag ensures, that the image is not auto-selected when the image version is not specified.
** After testing, remove the [literal]``Inactive`` flag so the image can be auto-selected in the [literal]``HWTYPE`` group.

> ** The [literal]``Inactive`` flag can be accessed through the API. i

Author: thank you

This file describes how to deploy a new image on selected terminals first, test it, and then deploy it to all terminals.

> The hardware type group (e.g [literal]``HWTYPE``) is used for booting new terminals, while already deployed terminals can be moved to any group that provides the {saltboot} formula. 

Reviewer: ```suggestion
The hardware type group (i.e. [literal]``HWTYPE`` group) is used for booting new terminals, while already deployed terminals can be moved to any group that provides the {saltboot} formula. 
```
